include/lbann/utils/protobuf/impl.hpp

```diff
@@ -120,6 +121,27 @@ auto get_repeated_field_ref(google::protobuf::Message const& msg,
 } // namespace protobuf
 } // namespace lbann
 
+template <typename T, typename ContainerT>
+void lbann::protobuf::assign_to_repeated(google::protobuf::RepeatedField<T>& field,
+                        ContainerT const& values)
+{
+  /** @todo Change to Assign if older versions of protobuf are no
+   *  longer supported.
+   */
+  field.Clear();
+  field.Add(begin(values), end(values));
+}
+
+template <typename ContainerT>
+std::string lbann::protobuf::to_space_sep_string(ContainerT const& values)
+{
+  std::string combined;
+  for (auto const& value : values)
+    combined += (value + " ");
+
+  return combined;
+}
```

Comment:
```suggestion
/** @brief Stringify the container with spaces between elements.
 *
 *  The `value_type` of the container must support stream output
 *  (e.g., `vector` or `set`, but not `map`).
 */
template <typename ContainerT>
std::string lbann::protobuf::to_space_sep_string(ContainerT const& values)
{
  if (values.empty())
    return "";
  std::ostringstream oss;
  oss << *cbegin(values);
  for (auto i = next(cbegin(values)); i != cend(values); ++i)
    oss << " " << *i;
  return oss.str();
}
```
We should be careful to allow containers of streamable non-strings (your impl will only work for strings). The stream operator is easier to extend than `to_string` for user types.